packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -708,6 +708,8 @@ class _AppBarState extends State<AppBar> {
     Scaffold.of(context).openEndDrawer();
   }
 
+  bool? previouslyImplyLeadingPopButton;
```

Comment:
Maybe call this `hadBackButtonWhenRouteWasActive` and include a comment that explains what the variable means when it's null, as well as true/false.